### PR TITLE
WIP [OCPCLOUD-1207] clean up carried patches for OpenShift

### PR DESCRIPTION
**Pull request by elmiko**
This is a large change that will partially rewrite our history of carried commits.

OpenShift infra commits
---
The following commits are all related directly to OpenShift infrastructure artifacts, they have all been squashed together into a single commit:

commit | description
---------- | --------------
d4828268c | UPSTREAM: <carry>: Updating vertical-pod-autoscaler images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/691e628254f318ce56efda5edc7448ec743c37b8/images/vertical-pod-autoscaler.yml
b6f2dd3e1 | UPSTREAM: <carry>: Updating atomic-openshift-cluster-autoscaler images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/691e628254f318ce56efda5edc7448ec743c37b8/images/atomic-openshift-cluster-autoscaler.yml
c2a34a75a | Updating .ci-operator.yaml `build_root_image` from openshift/release
1f2e3f182 | UPSTREAM: <carry>: openshift: update testing scripts to use go 1.16
5c0778df1 | UPSTREAM: <carry>: openshift: Fix build with Go 1.16
c1171a8d3 | UPSTREAM: <carry>: Updating vertical-pod-autoscaler builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/e3b61ff1dde2bfc2bc7f2b5efdd155d1d3299cd7/images/vertical-pod-autoscaler.yml
ba523e1aa | UPSTREAM: <carry>: VPA: Add Chen to approvers
78a33cde3 | UPSTREAM: <carry>: VPA: Update Dockerfiles to use new registry
5bcb8e2b3 | UPSTREAM: <carry>: Updating vertical-pod-autoscaler builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/5a1293dd0f380abf50c12d65c36655486d7745d0/images/vertical-pod-autoscaler.yml
981e40c5f | UPSTREAM: <carry>: Updating atomic-openshift-cluster-autoscaler builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/5a1293dd0f380abf50c12d65c36655486d7745d0/images/atomic-openshift-cluster-autoscaler.yml
3b9c2c9bc | UPSTREAM: <carry>: openshift: add component and subcomponent to OWNERS
a7c5f843a | UPSTREAM: <carry>: Updating vertical-pod-autoscaler builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/f82a216a6a3707b80a635bace9367f1a8288b7a7/images/vertical-pod-autoscaler.yml
532b0d2d9 | UPSTREAM: <carry>: Updating atomic-openshift-cluster-autoscaler builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/f82a216a6a3707b80a635bace9367f1a8288b7a7/images/atomic-openshift-cluster-autoscaler.yml
a70b078a0 | UPSTREAM: <carry>: openshift: Updating vertical-pod-autoscaler/Dockerfile.rhel baseimages to match ocp-build-data config
5983ad7f6 | UPSTREAM: <carry>: openshift: Updating images/cluster-autoscaler/Dockerfile.rhel baseimages to match ocp-build-data config
1783db6f7 | UPSTREAM: <carry>: openshift: Add vertical-pod-autoscaler/Dockerfile.rhel to match build configuration in ocp-build-data
e3cc25340 | UPSTREAM: <carry>: openshift: Add images/cluster-autoscaler/Dockerfile.rhel to match build configuration in ocp-build-data
d75d2ad30 | UPSTREAM: <carry>: openshift: update cluster-autoscaler OWNERS
5de16e42e | UPSTREAM: <carry>: Bump scripts go version to 1.13 and update path to cloudprovider/clusterapi
40e89c0a7 | UPSTREAM: <carry>: openshift: Extend makefile with 'make goimports' target
afe94421f | UPSTREAM: <carry>: Fix git commit message verification script
9c8fe7b41 | UPSTREAM: <carry>: openshift: Switch builds to use Go 1.12
96dbaae71 | UPSTREAM: <carry>: openshift: create git history verification script
ba9a9f67c | UPSTREAM: <carry>: openshift: Add fmt, lint, vet scripts/Makefile
5491a5afb | UPSTREAM: <carry>: openshift: cluster-autoscaler.spec: set golang_version to 1.10
43f2f10c0 | UPSTREAM: <carry>: openshift: cluster-autoscaler.spec: bump golang_version to 1.10.4
e51fb711c | UPSTREAM: <carry>: openshift: Fix spec file to be consistent
0e5a06ea3 | UPSTREAM: <carry>: openshift: Bump embedded tools
c0a87cc53 | UPSTREAM: <carry>: openshift: Fix the spec and hack scripts so the package can be built in both build systems
ec99d631c | UPSTREAM: <carry>: openshift: Add openshift/release Makefile and hack scripts
7cadbe115 | UPSTREAM: <carry>: openshift: Add dockerfile for cluster autoscaler.
30b6ee0de | UPSTREAM: <carry>: openshift: Add spec file for cluster-autoscaler.

Scale from zero commits
---
The following commits are all related to scale from zero behavior and have been squashed into a single "scale from zero" commit.

commit | description
---------- | --------------
457274d86 | UPSTREAM: <carry>: openshift: Add tests for nodegroup TemplateNodeInfo
38c7e5102 | UPSTREAM: <carry>: openshift: Copy standard labels from an existing node if one exists
6242cde8c | UPSTREAM: <carry>: openshift: Ensure stable versions of generic labels are also included
a43c8c2f8 | UPSTREAM: <carry>: Convert the mem value consistently with other providers
aea75a941 | UPSTREAM: <carry>: openshift: Implement scale from zero

Carried commits
---
These commits are carried through as they have special meaning for OpenShift and would be difficult, or inappropriate, to merge into the upstream.

commit | description
---------- | --------------
7411f238a | UPSTREAM: <carry>: openshift: Rename FailureMessage to ErrorMessage
32f60826c | UPSTREAM: <carry>: openshift: Revert "Adding config for event filtering"
a2285158d | UPSTREAM: <carry>: openshift: Set default annotations, and cluster name label to use machine.openshift.io

The following commit is carried for now but will be dropped when we rebase to 1.22

commit | description
---------- | --------------
49ad15cb4 | UPSTREAM: 4039: openshift: updating CA version to 1.21.0

Special commit
---
This commit was created to fix issues that had come up during previous rebase efforts. It contained fixups for scale from zero related issues and issues related to the OpenShift specific group names. This commit has been split and squashed into the scale from zero commit and the "Set default annotations..." commit.

commit | description
---------- | --------------
962d18ab7 | UPSTREAM: <carry>: openshift: add fixes for unstructured scalable resources and node group discovery

### CODE REVIEW

**Comment by elmiko** [comment]
@JoelSpeed @enxebre , i will most likely need your help reviewing this. i'm not sure that i've formatted the PR correctly as it rewrites the history. i might need to re-craft this in a similar fashion as our rebase PRs.

**Comment by elmiko** [comment]
/retitle WIP [OCPCLOUD-1207] clean up carried patches for OpenShift

**Comment by elmiko** [comment]
yeah, i crafted this wrong, need to create the history like we do for a rebase. i'll fix it up tomorrow, but the commits here are what we will carry over upstream going forward.

File: OWNERS [hunk 1]
```diff
@@ -1,8 +1,15 @@
 approvers:
-- mwielgus
-- maciekpytel
-- bskiba
+- enxebre
+- frobware
```
Comment (JoelSpeed) [inline OWNERS:3]: Perhaps while we are doing this, we should update the list of approvers, this is very out of date isn't it?

**Review by JoelSpeed — COMMENTED** [review]
Commits look more sensible to me, thanks!

File: OWNERS (hunk 1 above)
Comment (elmiko) [reply OWNERS:3]: good point

**Comment by elmiko** [comment]
i'm having trouble crafting this PR in a way that github will not barf on it. i'm going to poke around a little more, but given where we are in the release cycle, i wonder if maybe we should combine this rework with the upcoming rebase to 1.22?

wdyt @JoelSpeed @enxebre ?

**Comment by JoelSpeed** [comment]
If you believe that will be easier, I'm happy for it to wait til then, will wait for Alberto's ack though

**Comment by enxebre** [comment]
>i'm having trouble crafting this PR in a way that github will not barf on it. i'm going to poke around a little more, but given where we are in the release cycle, i wonder if maybe we should combine this rework with the upcoming rebase to 1.22?

Yes, this exercise needs to be done as part of the every release rebase. We should have a PR for any change downstream done during a release cycle and we should not carry additional commits from one release to another one unless there's a reason to.

**Comment by elmiko** [comment]
thanks for the input guys, i am going to close this and keep the branch around. i will use these new pruned commits as our carries for 1.22.

**Pull request closed without merging**